python/llm/src/ipex_llm/transformers/lookup.py

```diff
@@ -54,10 +57,12 @@ def generate(
     streamer: Optional["BaseStreamer"] = None,
     **kwargs,
 ):
+    device_name = get_xpu_device_type(inputs)
     lookahead = kwargs.pop("lookahead", None)
     perf_mode = os.environ.get("IPEX_LLM_PERFORMANCE_MODE", None)
     if perf_mode == "1" and lookahead is None:
-        lookahead = 2  # default to 2 now
+        if device_name != 'mtl' or inputs.shape[-1] >= PERFORMANCE_MODE_LOOKUP_INPUT_THRESHOLD:
```

Comment:
why `device_name != 'mtl'`?

Reply:
As previous tested data on Arc shows no slowdown for lookup on Orca & CNN datasets. Should we also add this threshold for all devices?

Reply:
> As previous tested data on Arc shows no slowdown for lookup on Orca & CNN datasets. Should we also add this threshold for all devices?

Yes, let's add it for all device for now

Reply:
Thank you :) Have updated.